code/game/objects/items/radio/headset.dm

```diff
@@ -220,8 +220,11 @@ GLOBAL_LIST_INIT(channel_tokens, list(
 		camera.toggle_cam(null, FALSE)
 	if(wearer.mind && wearer.assigned_squad && !sl_direction)
 		enable_sl_direction()
+	add_minimap()
 	to_chat(wearer, "<span class='notice'>You toggle the Squad HUD on.</span>")
 	playsound(loc, 'sound/machines/click.ogg', 15, 0, 1)
+	var/datum/action/minimap/mini = new
+	mini.give_action(wearer)
```

Comment:
You are already doing this in add_minimap()

Reply:
fixed